onnxscript/ir/_core.py

```diff
@@ -1187,18 +1189,18 @@ def __init__(
         self._version: int | None = version
         self._metadata: _metadata.MetadataStore | None = None
         self._metadata_props: dict[str, str] | None = metadata_props
-        self._graph: Graph | Function | None = graph
+        # _graph is set by graph.append
+        self._graph: Graph | None = None
+        # Add the node to the graph if graph is specified
+        if graph is not None:
+            graph.append(self)
```

Comment:
Doesn't this alter the behavior (wrt what was happening previously)? Trying to understand what happens during optimization/rewriting (now vs. previously): I thought we had the ability to process a node N, and replace it with say new nodes M1, M2 ... and that they get inserted in the place where N was. Trying to reconcile how the above `append` interacts with that?

Reply:
I think this should be the same behavior as before (only moving the lines from https://github.com/microsoft/onnxscript/pull/2183/files#diff-33a5ea7182ac2a3ba4127e9fea3cd2db2dbe54810994ca210727d28fccf2b5cfL1198-L1201)

When we insert by calling graph.insert*, node.graph will be set in the insert methods; when we create the replacement nodes `graph` is typically None so they do not get appended.